packages/melos/test/commands/bootstrap_test.dart

```diff
@@ -146,6 +146,126 @@ Generating IntelliJ IDE files...
       );
     });
 
+    test('properly compares the path changes on git references', () async {
+      final temporaryGitRepository = createTestTempDir();
+
+      await io.Process.run(
+        'git',
+        ['init'],
+        workingDirectory: temporaryGitRepository.absolute.path,
+      );
+
+      await createProject(
+        io.Directory('${temporaryGitRepository.absolute.path}/dependency1'),
+        const PubSpec(name: 'dependency'),
+      );
+
+      await createProject(
+        io.Directory('${temporaryGitRepository.absolute.path}/dependency2'),
+        const PubSpec(name: 'dependency'),
+      );
+
+      await io.Process.run(
+        'git',
+        ['add', '-A'],
```

Comment:
Empty directories won't be added to git, maybe create an empty file in both of the directories first?

Reply:
Ah right, I read `createDirectory` and not `createProject` when I read it this morning. :sweat_smile: 
The test passes locally for me too (linux), so permissions does sound like a good guess, but it doesn't look like it from the error message in the pipeline, if it doesn't fail silently earlier.

Reply:
I am going to try to reproduce the problem by using a container locally and running the same images. Maybe I can debug this way.

Reply:
Good idea, I re-ran the failing test with GitHub debug printing, but it didn't give anything

Reply:
@mateusfccp did you manage to find anything causing it?

Reply:
Hello, @spydon.

Sorry for the delay. I actually didn't get to reproduce the environment on my local machine. I tried to do it with `act`, but with no success.